diff -u -r ../cache/termux-api-package-0.59.1/termux-api.c ./termux-api.c
--- ../cache/termux-api-package-0.59.1/termux-api.c	2025-03-29 21:04:28.000000000 +0000
+++ ./termux-api.c	2025-09-18 23:37:36.300881974 +0000
@@ -60,7 +60,7 @@
     // the app process to deliver the intent.
     // - https://github.com/termux/termux-api/issues/638#issuecomment-1813233924
     int listenfd = -1;
-    #ifdef __ANDROID__
+    #if defined(__ANDROID__) && defined(RAFCODEPHI_SHARED_UID_FILESYSTEM_SOCKETS)
         if (android_get_device_api_level() < 34) {
             listenfd = socket(AF_UNIX, SOCK_STREAM|SOCK_CLOEXEC, 0);
         }
@@ -265,7 +265,7 @@
     child_argv[2] = "--user";
     child_argv[3] = "0";
     child_argv[4] = "-n";
-    child_argv[5] = "com.termux.api/.TermuxApiReceiver";
+    child_argv[5] = "com.termux.rafacodephi.api/com.termux.api.TermuxApiReceiver";
     child_argv[6] = "--es";
     // Input/output are reversed for the java process (our output is its input):
     child_argv[7] = "socket_input";
